### ISSUE TITLE
Feature request: Warn if copying with cat

### ISSUE THREAD

**Issue by ember91** [koalaman/shellcheck#2792]
#### For new checks and feature suggestions
- [ X ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ X ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

I found this in some code:

```sh
cat file1 > file2
```

Wouldn't it be a nice feature if shellcheck recommended this instead:

```sh
cp file1 file2
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Don't use 'cat' to copy files. Use cp instead:
```sh
cp file1 file2
```

**Comment by koalaman** [comment]
I see what you mean, this would look odd when a `cp` would do. There's no real harm done though, and redirecting `cat` has some potentially beneficial differences when it comes to propagating permissions, handling directories, bash special filenames, and the noclobber shell option, so a blanket policy doesn't seem worthwhile. Thanks for the suggestion!
